sys/parsoid.js

```diff
@@ -417,6 +439,18 @@ PSP.getFormat = function(format, restbase, req) {
     });
 
     if (req.headers && /no-cache/i.test(req.headers['cache-control'])) {
+        if (!self._okayToRerender(req)) {
+            // Still update the revision metadata.
+            self.getRevisionInfo(restbase, req);
+            throw new rbUtil.HTTPError({
```

Comment:
Wouldn't getting revinfo and throwing the error in parallel cancel the revinfo promise given that all of the operations are executed in the `P.try()` context?